src/controllers/HierarchicalFacetSearchController.ts

```diff
@@ -0,0 +1,38 @@
+import { IDynamicHierarchicalFacet, IDynamicHierarchicalFacetValue } from '../ui/DynamicHierarchicalFacet/IDynamicHierarchicalFacet';
+import { FacetSearchType, IFacetSearchRequest } from '../rest/Facet/FacetSearchRequest';
+import { IFacetSearchResponse } from '../rest/Facet/FacetSearchResponse';
+import { flatten } from 'underscore';
+
+type Path = string[];
+
+export class HierarchicalFacetSearchController {
```

Comment:
This class is for the most part a copy of [`FacetSearchController`](https://github.com/coveo/search-ui/blob/master/src/controllers/FacetSearchController.ts). The differences are mostly in the `search` method.